packages/sanity/src/structure/panes/documentList/useDocumentListSort.ts

```diff
@@ -0,0 +1,29 @@
+import {useMemo} from 'react'
+import {useSchema} from 'sanity'
+import {type DocumentListPaneNode} from 'sanity/structure'
+
+import {EMPTY_RECORD} from './constants'
+import {applyOrderingFunctions, findStaticTypesInFilter, useShallowUnique} from './helpers'
+import {type SortOrder} from './types'
+
+export const useDocumentListSort = (pane: DocumentListPaneNode, sortOrderRaw?: SortOrder) => {
+  const schema = useSchema()
+  const {options} = pane
+  const {filter, params: paneParams} = options
+  const params = useShallowUnique(paneParams || EMPTY_RECORD)
+  const typeName = useMemo(() => {
+    const staticTypes = findStaticTypesInFilter(filter, params)
+    if (staticTypes?.length === 1) return staticTypes[0]
+    return null
+  }, [filter, params])
```

Comment:
I wonder if we need this in here or we could refactor this to receive the `typeName` as a parameter. 
I think `typeName` is accessible and already calculated in both `DocumentSheetList` and in `DocumentListPane` where this hook is used.

Reply:
The `typename` used in `DocumentListPane` could be any typename though? That list supports rendering multiple different types, so I'm unsure how to replicate this there. I do agree regarding `DocumentSheetList` however

Reply:
sorry for the late reply on this.
The `sortWithOrderingFn` in the hook and in the previous implementation in `DocumentSheetList` uses the `typeName` which results in the first staticType found.

So I was thinking, given DocumentListPane is already resolving the typeName [here](https://github.com/sanity-io/sanity/blob/edx-1461-sort-sheet-list/packages/sanity/src/structure/panes/documentList/DocumentListPane.tsx#L49-L54) 

```ts 
// DocumentListPane:69
 const sortWithOrderingFn = useDocumentListSort(typeName, sortOrderRaw)

// DocumentSheetListPane:146
const sortWithOrderingFn = useDocumentListSort(documentSchemaType.name, sortOrderRaw)

```

And then `useDocumentListSort` will only require this:
```ts
export const useDocumentListSort = (typeName: string | null, sortOrderRaw?: SortOrder) => {
  const schema = useSchema()

  const sortWithOrderingFn = useMemo(
    () =>
      typeName && sortOrderRaw
        ? applyOrderingFunctions(sortOrderRaw, schema.get(typeName) as any)
        : sortOrderRaw,
    [schema, sortOrderRaw, typeName],
  )

  return sortWithOrderingFn
}
```

Reply:
dohh! Right you are - that makes sense. I hadn't realised I'd now duplicated this logic. Thanks for realising!

- [ ] Simplify `useDocumentListSort` by just passing in `typename` directly